src/db/h2/lrsql/h2/result.clj

```diff
@@ -0,0 +1,41 @@
+(ns lrsql.h2.result
+  "Process raw h2 results for parity with other backends"
+  (:require [com.yetanalytics.squuid :as squuid]
+            [lrsql.util :as u]))
+
+(defn query-last-statement-stored-result
+  "Extracts a timestamp suitable for last-statement-stored from an H2 query
+  result. If result is nil, returns nil."
+  [query-result]
+  (when-let [{:keys [id]} query-result]
+    {:lstored (-> id squuid/uuid->time u/time->str)}))
+
+(defn query-platform-frequency-result
+  "Given a result of all statement payloads, return the frequencies of each"
+  [query-result]
+  (map
+   (fn [[platform scount]]
+     {:platform platform
+      :scount   scount})
+   (frequencies
+    (map #(get-in % [:payload "context" "platform"] "unknown")
```

Comment:
Is "unknown" really the right word to use in the case the statement has no platform (as opposed to "none" or similar)?

Reply:
Good question. I came at it from the position that every statement _comes from a platform of some kind_ but maybe that's wrong. I'll read the spec language about it.

Reply:
[According to the spec](https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#requirements-10) platform only gets supplied for statements with an activity object, so not every statement, though I don't think we need to really differentiate in the query result.
I think `none` is fine I'll go with that.

Reply:
Cool. As a side note there could be issues if there's a platform literally named `none` (or previously `unknown`) but nothing to do about that. :shrug:

Reply:
True, though the meaning is the same either way, explicit or implicit haha